src/views/TradingCompetition/hooks/useGetCompetitionState.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react'
+import { useInitialBlock } from '../../../state/hooks'
+import { CompetitionState } from '../config'
+
+const useGetCompetitionState = () => {
+  const [targetBlock, setTargetBlock] = useState(null)
+  const [competitionState, setCompetitionState] = useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const currentBlock = useInitialBlock()
+
+  useEffect(() => {
+    const liveBlock = CompetitionState.LIVE.startBlock
+    const finishedBlock = CompetitionState.FINISHED.startBlock
+
+    if (currentBlock < liveBlock) {
+      setTargetBlock(liveBlock)
+      setCompetitionState(CompetitionState.REGISTRATION)
+    } else if (currentBlock < finishedBlock) {
+      setTargetBlock(finishedBlock)
+      setCompetitionState(CompetitionState.LIVE)
+    } else {
+      setTargetBlock(0)
+      setCompetitionState(CompetitionState.FINISHED)
+    }
+  }, [currentBlock])
+
+  useEffect(() => {
+    if (currentBlock > 0 && targetBlock && competitionState && isLoading) {
```

Comment:
`currentBlock` (from the `useInitialBlock` hook) takes a few ms to fetch, and without it the countdown data is incomplete (`useGetBlockCountdown` also relies on it). As a result I created an isLoading state until that value is greater than 0. 

Including the other values in this condition - `targetBlock` & `competitionState` might be OTT, but I included them anyway.